website/static/js/components/publicNodes.js

```diff
@@ -0,0 +1,344 @@
+                ctrl.paginators().push({
+                    text: '...',
+                    url: function() { }
+                });
+                for (i = ctrl.totalPages() - MAX_PAGES_ON_PAGINATOR_SIDE + 2; i <= ctrl.totalPages() - 1; i++) {
+                    ctrl.paginators().push({
+                        text: i,
+                        url: function() {
+                            ctrl.pageToGet(parseInt(this.text));
+                            if (ctrl.pageToGet() !== ctrl.currentPage()) {
+                                return _buildUrl(ctrl.pageToGet(), ctrl.user, ctrl.nodeType);
+                            }
+                        }
+                    });/* jshint ignore:line */
+                    // function defined inside loop
+                }
+            }
+            // two ellipses
+            else {
+                ctrl.paginators().push({
+                    text: '...',
+                    url: function() { }
+                });
+                for (i = parseInt(ctrl.currentPage()) - 1; i <= parseInt(ctrl.currentPage()) + 1; i++) {
+                    ctrl.paginators().push({
+                        text: i,
+                        url: function() {
+                            ctrl.pageToGet(parseInt(this.text));
+                            if (ctrl.pageToGet() !== ctrl.currentPage()) {
+                                return _buildUrl(ctrl.pageToGet(), ctrl.user, ctrl.nodeType);
+                            }
+                        }
+                    });/* jshint ignore:line */
+                    // function defined inside loop
+                }
+                ctrl.paginators().push({
+                    text: '...',
+                    url: function() { }
+                });
+            }
+            // last page
+            ctrl.paginators().push({
+                text: ctrl.totalPages(),
+                url: function() {
+                    ctrl.pageToGet(ctrl.totalPages());
+                    if (ctrl.pageToGet() !== ctrl.currentPage()) {
+                        return _buildUrl(ctrl.pageToGet(), ctrl.user, ctrl.nodeType);
+                    }
+                }
+            });
+            // next page
+            ctrl.paginators().push({
+                url: function() { return ctrl.nextPage(); },
+                text: '>'
+            });
+        }
+
+        return m('ul.list-group m-md', [
+            // Error message if the request fails
+            ctrl.failed ? m('p', [
+                'Unable to retrieve public ' + ctrl.nodeType + ' at this time. Please refresh the page or contact ',
+                m('a', {'href': 'mailto:support@osf.io'}, 'support@osf.io'),
+                ' if the problem persists.'
+            ]) :
+
+            // Show laoding icon while there is a pending request
+            ctrl.requestPending() ?  m('.ball-scale.ball-scale-blue.text-center', m('')) :
```

Comment:
I believe `ball-scale` is meant to be used for a single large area rather than multiple widgets on the page. `ball-pulse` might be more appropriate here. Is this correct, @caneruguz ?